sw/device/lib/dif/dif_keymgr.h

```diff
@@ -0,0 +1,656 @@
+   * can be used to generate an owner identity using an asymmetric KDF.
+   */
+  kDifKeymgrStateOwnerRootKey,
+  /**
+   * Disabled state.
+   *
+   * This is a terminal state where key manager is no longer operational. At
+   * this state, the secret value of key manager is a random value.
+   */
+  kDifKeymgrStateDisabled,
+} dif_keymgr_state_t;
+
+/**
+ * Creates a new handle for key manager.
+ *
+ * This function does not actuate the hardware and must be called to initialize
+ * the handle that must be passed to other functions in this library in each
+ * boot stage. A typical usage of this library during different secure boot
+ * stages is as follows:
+ *
+ * - In Mask ROM:
+ *   - Create a new handle: `dif_keymgr_init()`.
+ *   - Configure hardware: `dif_keymgr_configure()`.
+ *   - Initialize state: `dif_keymgr_advance_state()`,
+ *   `dif_keymgr_get_status_codes()`, `dif_keymgr_get_state()`.
+ *   - Advance state: `dif_keymgr_advance_state()`,
+ *     `dif_keymgr_get_status_codes()`, `dif_keymgr_get_state()`.
+ * - In subsequent boot stages, i.e. ROM_EXT, BL0, kernel:
+ *   - Create a new handle: `dif_keymgr_init()`.
+ *   - Generate keys and/or identity seeds:
+ *     `dif_keymgr_generate_versioned_key()`,
+ *     `dif_keymgr_generate_identity_seed()`, `dif_keymgr_get_status_codes()`.
+ *   - Read output (if applicable): `dif_keymgr_read_output()`.
+ *   - Advance state: `dif_keymgr_advance_state()`,
+ *     `dif_keymgr_get_status_codes()`, `dif_keymgr_get_state()`.
+ *
+ * @param params Hardware instantiation parameters.
+ * @param[out] keymgr Out-param for the initialized handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_keymgr_result_t dif_keymgr_init(dif_keymgr_params_t params,
+                                    dif_keymgr_t *keymgr);
+
+/**
+ * Configures key manager with runtime information.
+ *
+ * This function should need to be called once for the lifetime of `keymgr`.
+ *
+ * @param keymgr A key manager handle.
+ * @param config Runtime configuration parameters.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_keymgr_result_t dif_keymgr_configure(const dif_keymgr_t *keymgr,
+                                         dif_keymgr_config_t *config);
+
+/**
+ * Parameters for a key manager state.
+ */
+typedef struct dir_keymgr_state_params {
+  /**
+   * This value is used by key manager to derive secret values and can be either
+   * a value that represents the contents of a boot stage, e.g. a (truncated)
+   * hash, or a tag.
+   *
+   * If it is a hash, changes in a boot stage will change the secret value, and
+   * consequently the versioned keys and identity seeds generated at subsequent
+   * boot stages. If it is a tag, those secret values, versioned keys, and
+   * identity seeds will be preserved across updates of the boot stage as long
+   * as the tag remains the same.
+   */
+  uint8_t binding_value[16];
```

Comment:
@alphan, @tjaychen 

If we are using `binding_value` as the `ROMExtDescriptor`,  would it be possible to make the binding value `256B` so that we can use hash values if needed?

Reply:
Done.